File: Utils/Javascript/StrategyPerformanceMetrics.js
// StrategyPerformanceMetrics.js

import { Decimal } from 'decimal.js';

/**
 * Class to track performance metrics of different strategies.
 */
class StrategyPerformanceMetrics {
    /**
     * Initializes the StrategyPerformanceMetrics instance.
     */
    constructor() {
        this.successes = 0;
        this.failures = 0;
        this.profit = new Decimal(0);
        this.avg_execution_time = 0.0;
        this.success_rate = 0.0;
        this.total_executions = 0;
    }
}

export default StrategyPerformanceMetrics;

File: utils/strategyperformancemetrics.py
class StrategyPerformanceMetrics:
    successes: int = 0
    failures: int = 0
    profit: Decimal = Decimal("0")
    avg_execution_time: float = 0.0
    success_rate: float = 0.0
    total_executions: int = 0

@dataclass